<launch>

  <arg name="image_path"		default="triangulate" />
  <arg name="vis"			default="false" />
  <arg name="board_ip" 			default="127.0.0.1" />  
  <arg name="exposures" 		default="1" />
  <arg name="has_left_cam" 		default="true" />
  <arg name="has_right_cam" 		default="true" />
  <arg name="has_color_cam" 		default="true" />
  <arg name="stereo_cam_topic"  	default="vrm3dvision" />
  <arg name="pc_color_option"   	default="none" />
  <arg name="remove_outliers"   	default="true" />
  <arg name="remove_dominant_plane"   	default="true" />
  <arg name="certainty_type"	default="0" />
  <arg name="layers"		default="-1" />

  <arg name="point_cloud_topic" default="point_cloud" />
  <arg name="point_cloud_trimmed_topic" default="point_cloud_trimmed" />

        <group ns="/">
               <node pkg="vrm3dvision" type="vrm_image_proc" name="VrmImageProcNode" output="screen">
			<param name="vrm_board_ip" value="$(arg board_ip)" />
			<param name="has_left_camera" value="true" />
			<param name="has_right_camera" value="true" />
			<param name="has_color_camera" value="true" />
			<param name="stereo_camera_topic" value="vrm_stereo_camera" />
			<param name="color_cam_topic" value="color_cam" />
			<param name="visualization" value="$(arg vis)" />
			<param name="no_of_layers_structured_light" value="$(arg layers)" />
			<param name="point_cloud_pub_topic" value="$(arg point_cloud_topic)" />			
			<param name="point_cloud_trimmed_pub_topic" value="$(arg point_cloud_trimmed_topic)" />
			<param name="publish_trimmed_point_cloud" value="true" />			
			<param name="z_trim_distance" value="0.40" />			
			<param name="calibration_path" value="$(arg image_path)" />
			<param name="pc_color_option" value="$(arg pc_color_option)" />
 			<param name="remove_outliers" value="$(arg remove_outliers)" /> 
			<param name="remove_dominant_plane" value="$(arg remove_dominant_plane)" />	
			<param name="certainty_type" value="$(arg certainty_type)" />
 			<param name="mean_k" value="20" />
 			<param name="std_dev_thresh" value="1.5" />
                </node>		
		<node pkg="vrm3dvision" type="pcl_pointcloud_viewer" name="PCLPCviewer" output="screen">
			<param name="point_cloud_subscriber_topic" value="$(arg point_cloud_topic)" />
		</node> 

		<node pkg="vrm3dvision" type="pcl_pointcloud_viewer" name="PCLPCviewerTrimmed" output="screen">
			<param name="point_cloud_subscriber_topic" value="$(arg point_cloud_trimmed_topic)" />
		</node> 

		<node pkg="vrm3dvision" type="zmq_image_publisher_node" name="ZMQimagePublisher" output="screen" >
			<param name="image_folder" value="$(arg image_path)" />			
			<param name="has_left_camera" value="$(arg has_left_cam)" />
			<param name="has_right_camera" value="$(arg has_right_cam)" />
			<param name="has_color_camera" value="$(arg has_color_cam)" />	
			<param name="no_of_layers_structured_light" value="$(arg layers)" />
			<param name="publish_on_startup" value="false" />
			<param name="exposures" value="$(arg exposures)" type="str"/>
		</node>	

		<node pkg="rqt_service_caller" type="rqt_service_caller" name="RqtServiceCaller" />

        </group>
</launch>
